daft/table/table_io.py

```diff
@@ -527,3 +520,185 @@ def file_visitor(written_file, i=i):
         for c_name in partition_values.column_names():
             data_dict[c_name] = partition_values.get_column(c_name).take(partition_idx_series)
     return MicroPartition.from_pydict(data_dict)
+
+
+def coerce_pyarrow_table_to_schema(pa_table: pa.Table, input_schema: pa.Schema) -> pa.Table:
```

Comment:
I find this very frustrating of Arrow where you cannot just cast the table. I did already quite a bit of work on this on the Arrow side, but is still not complete :(